### ISSUE TITLE
Nice little app!

### ISSUE THREAD

**Issue by Nisse46** [randyrants/sharpkeys#235]
Hi,
I.m trying to create a "sleeep" button for my wifes new keyboard.
Remapping any of the keys to become a sleep button seems impossible.
The key I remap looses it's original function but no computer sleep.
Why?
Rgds
Nils

**Comment by SumatraPeter** [comment]
Have you set Windows to actually sleep when the sleep button is pressed?

https://www.digitalcitizen.life/change-shut-down-button-log-lock-restart-sleep-etc

If you don't see the option to customize your sleep button's action in the old Windows Control Panel (perhaps because the system doesn't actually have a sleep button), you'll have to do it another way (for example, create a shortcut to a program that sends the PC to sleep).

**Comment by randyrants** [comment]
Also, sleep is a strange key on most laptops and keyboards.  It's sometimes intercepted by hardware which means it never gets passed to Windows, so Windows cannot map it.  In other cases, a machine will ignore the Sleep scan code because it only responds to the hardware one. 

It's a try and see sort of thing.

**Comment by Nisse46** [comment]
I have an old Microsoft keyboard with a "sleep key". The computer is a desktop with Win10 pro. If i enter settings in Windows, energy, behaviour when sleep key is pressed, sets it to no action, the sleep button on the keyboard stops working. So somehow windows must recognise that key. If I choose turn off monitor the monitor goes off.
Sharp Key can't recognise any output from the keyboard in those cases. 
Really odd?? Does Sharp Key filter codes that are unexpected? 
Is there any program that's logging Everything that the keyboard produces?
Rgds
Nissse

**Comment by vaifrax** [comment]
I also tried to remap any key to put the PC to sleep. The key loses it's original function, but nothing happens. The registry entry looks fine, though.
Also, on another Windows 10 machine it did not work.
Also, the Power (down) code does nothing.
Also, the break code (DF_E0) or the Set 2 code (3F_E0) did not work (done via changing in regedit).

It seems to be a Windows problem, not a Sharpkeys problem, but somebody here can help. Does Windows maybe expect both the make and break codes?
@SumatraPeter I configured my case power button to sleep but I'm guessing this has nothing to do with keyboard input.

**Comment by SumatraPeter** [comment]
I finally had the opportunity to test on a desktop with an old IBM 101-key KB. Set F12 to Sleep but it did nothing at all after remapping. I'm not really sure whether the Sleep button on a new KB will even work with the same PC.

@vaifrax: No (would be weird if it did).

**Comment by ScottAwesome** [comment]
I wanted to drop in a thank you here. Not an issue, but definitely in theme with the issue's title!

**Comment by Contramalgra** [comment]
@ScottAwesome It's on his main website, https://www.randyrants.com/donate/

**Comment by yz778** [comment]
Hi, adding to this thread - thank you so much for making SharpKeys!

I'm using a portable keyboard that has only has a num pad, and it was driving me nuts b/c ALT+arrow keys produced symbols even when Numlock was off. That made coding awful since using ALT+UP/DN to move lines produced little symbols all over the place.

Was about to give up and buy a new keyboard until I stumbled on this tool. Thank you thank you!

**Comment by Juckelfunk** [comment]
Hey, I just wanted to let you know that I successfully remapped the Alt Gr key.
Thanks a lot for this app, it's absolutely great! Now i can finally properly use my Unicomp Model M. (They have the right Alt and Windows key switched for some reason).

I have a German keyboard btw, I don't know if that is important.

**Comment by JimJettuman** [comment]
Is it possible to remap the windows keyboard with this application so it works with regional versions of Commodore64 emulator Vice? see this thread for reference: https://www.lemon64.com/forum/viewtopic.php?t=71226

I will try it out myself, but was hoping someone had explored similar uses already.

Thanks!

**Comment by randyrants** [comment]
Sadly, the Windows tech that is used to remap things is not aware of any specific application, so this product cannot help.  That said, give Microsoft PowerToys a try: I know they have a more active remapping but I don't remember if they are product by product.

**Comment by Lozzybet** [comment]
Hi, 
Thanks for creating this handy little app, it's wonderful.
I've a new Microsoft Ergonomic keyboard and it has two new keys - Office 365, and Emojis.  
I never use them, but the Office 365 is next to the right Alt key and I often accidentally hit it and it gets in the way, so I'd like to disable it.  But when I try it says it's not recognised. It's code is E0_205B.

Is it possible to add this so it can be disabled, please?

I tried to add the code for the Emoji button for reference, but it wasn't detected.

Many thanks!

**Comment by DreymaR** [comment]
> I've a new Microsoft Ergonomic keyboard and it has two new keys - Office 365, and Emojis. I never use them, but the Office 365 is next to the right Alt key and I often accidentally hit it and it gets in the way, so I'd like to disable it. But when I try it says it's not recognised. It's code is E0_205B.
> 
> Is it possible to add this so it can be disabled, please?
> 
> I tried to add the code for the Emoji button for reference, but it wasn't detected.

According to SharpKeys' documentation, such three-byte key codes cannot be remapped. You may be out of luck.

If you pry the key cap carefully off the offending key, maybe you can do something with it that way? Such as insulating its contact, or, failing that, simply leaving the key cap off.

**Comment by randyrants** [comment]
> > I've a new Microsoft Ergonomic keyboard and it has two new keys - Office 365, and Emojis. I never use them, but the Office 365 is next to the right Alt key and I often accidentally hit it and it gets in the way, so I'd like to disable it. But when I try it says it's not recognised. It's code is E0_205B.
> > Is it possible to add this so it can be disabled, please?
> > I tried to add the code for the Emoji button for reference, but it wasn't detected.
> 
> According to SharpKeys' documentation, such three-byte key codes cannot be remapped. You may be out of luck.
> 
> If you pry the key cap carefully off the offending key, maybe you can do something with it that way? Such as insulating its contact, or, failing that, simply leaving the key cap off.

Indeed, SharpKeys cannot help with these two keys.  However, you can install the Microsoft Keyboard software to remap some of these keys.  I'm pretty sure the more recent version would handle both of these keys, but I haven't personally tried.  I just ignore them (and use WIN+; for emoji) :)

**Comment by vgdh** [comment]
Thank you so much! It totally reshapes my work with a keyboard ❤

**Comment by randyrants** [comment]
> It gives this command to run in PowerShell, and it simply disables it: REG ADD HKCU\Software\Classes\ms-officeapp\Shell\Open\Command /t REG_SZ /d rundll32

Thanks @Lozzybet - I've added this bit to the FAQ!

**Comment by rainlights** [comment]
Thank you for this app. I write and translate for a living, but I never was a great typer. Especially the quotation mark was always a nuisance for me: it's next to the "0" on German keyboards. I switched it with the absolutely useless "^" key (which for some reason is featured prominently to the upper left). So now "?" is next to "!", and it made my life so much easier.
